File: backends/spring-kotlin/src/main/kotlin/dev/stackverse/backend/account/UserAccount.kt
package dev.stackverse.backend.account

import com.fasterxml.jackson.annotation.JsonValue
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant

enum class UserAccountStatus(@get:JsonValue val wire: String) {
    ACTIVE("active"),
    BLOCKED("blocked"),
}

/** App-level account, lazily provisioned from JWTs (SPEC rule 16) — identity itself is Keycloak's. */
@Entity
@Table(name = "user_accounts")
class UserAccount(
    @Id
    val username: String,
    val firstSeen: Instant,
    var lastSeen: Instant,
    @Enumerated(EnumType.STRING)
    var status: UserAccountStatus,
    var blockedReason: String?,
)

File: backends/dotnet/src/StackverseBackend/Accounts/UserAccount.cs
namespace StackverseBackend.Accounts;

public enum UserAccountStatus
{
    Active,
    Blocked,
}

/// <summary>App-level account, lazily provisioned from JWTs (SPEC rule 16) — identity itself is Keycloak's.</summary>
public class UserAccount
{
    public required string Username { get; init; }
    public DateTime FirstSeen { get; init; }
    public DateTime LastSeen { get; set; }
    public UserAccountStatus Status { get; set; }
    public string? BlockedReason { get; set; }
}
